src/main/java/org/sciborgs1155/robot/Hopper/Hopper.java

```diff
@@ -0,0 +1,54 @@
+package org.sciborgs1155.robot.Hopper;
+
+import static org.sciborgs1155.robot.Hopper.HopperConstants.*;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import monologue.Logged;
+import monologue.Monologue.LogBoth;
+import org.sciborgs1155.robot.Robot;
+
+public class Hopper extends SubsystemBase implements Logged {
+
+  final PIDController pid = new PIDController(kp, ki, kd);
+  public final HopperIO hopper = Robot.isReal() ? new RealHopper() : new SimHopper();
+
+  @LogBoth public double targetSpeed = pid.getSetpoint();
```

Comment:
same thing with the isAtTarget(). Although actually, might as well just log the pidController and then you don't need this at all.